### PR TITLE
[Model monitoring] User permissions hardening

### PR DESCRIPTION
**Pull request by Michaelliv**
The purpose of this pull request is to ensure user permissions also affect the model endpoints API.

PR change log:
* `V3IO_ACCESS_KEY` is now expected to be passed via a `X-V3io-Session-Key` request header.
* `V3IO_API` is now computed via a `get_secrets` functions that relies on already-existing environment params.
* `V3IO_FRAMESD` is now computed via a `get_secrets` functions that relies on already-existing environment params.
* Endpoint KV and TSDB records are now partitioned by the project's name.
* ModelEndpoint `id` is now computed as part of the initialization of a ModelEndpoint instance.
* Tests are fixed accordingly to new API changes.

### CODE REVIEW

File: mlrun/api/api/endpoints/model_endpoints.py [hunk 1]
```diff
@@ -363,9 +393,31 @@ def _verify_endpoint(project, endpoint_id):
...
+    else:
+        v3io_api = v3io_api.replace("tcp", "http")
+
+    v3io_framesd = environ.get("FRAMESD_PORT_8081_TCP")
+    if not v3io_framesd:
+        raise MLRunPreconditionFailedError(
+            "Environment missing 'V3IO_WEBAPI_PORT_8081_TCP' parameter."
+        )
+    else:
+        v3io_framesd = v3io_framesd.replace("tcp", "http")
```
Comment (Hedingber) [inline mlrun/api/api/endpoints/model_endpoints.py:417]: This counts on the fact that there is a svc (k8s service) named `v3io-webapi` - the svc name is originated from the app service name, we don't want to this coupling. (same for framesd)
Please simply... started describing what I want but understood it's a bit complicated - done that in https://github.com/mlrun/mlrun/pull/691 - so here you'll simply do `config.httpdb.v3io_api/v3io_framesd`

Comment (Hedingber) [inline mlrun/api/api/endpoints/model_endpoints.py:417]: This counts on the fact that there is a svc (k8s service) named `v3io-webapi` - the svc name is originated from the app service name, we don't want to this coupling. (same for framesd)
Please simply... started describing what I want but understood it's a bit complicated - done that in https://github.com/mlrun/mlrun/pull/691 - so here you'll simply do `config.httpdb.v3io_api/v3io_framesd`

File: mlrun/api/api/endpoints/model_endpoints.py [hunk 2]
```diff
@@ -363,9 +393,31 @@ def _verify_endpoint(project, endpoint_id):
...
+
+    return {
+        "V3IO_ACCESS_KEY": access_key,
+        "V3IO_API": v3io_api,
+        "V3IO_FRAMESD": v3io_framesd,
+    }
```
Comment (Hedingber) [inline mlrun/api/api/endpoints/model_endpoints.py:423]: Use the constants you defined

File: mlrun/api/schemas/model_endpoints.py [hunk 3]
```diff
@@ -30,6 +32,18 @@ class ModelEndpoint(BaseModel):
...
+    id: Optional[str] = None
+
+    def __init__(self, **data: Any):
+        super().__init__(**data)
+        self.id = self.create_endpoint_id()
+
+    def create_endpoint_id(self):
+        endpoint_unique_string = (
+            f"{self.spec.function}_{self.spec.model}_{self.metadata.tag}"
+        )
```
Comment (Hedingber) [inline mlrun/api/schemas/model_endpoints.py:44]: I guess you should check that all of this fields are filled no ?

**Review by Hedingber — CHANGES_REQUESTED** [review]
Just reminding (not urgent) that at some point you'll need to convert the tests here to system tests (https://github.com/mlrun/mlrun/pull/691 prepare the land for that), and ideally add some unit tests

File: mlrun/api/api/endpoints/model_endpoints.py [hunk 4]
```diff
@@ -140,11 +150,13 @@ def list_endpoints(
     Or by using a `,` (comma) seperator:
     `api/projects/{project}/model-endpoints/?label=mylabel=1,myotherlabel=2`
     """
-
-    client = get_v3io_client()
+    secrets = get_model_endpoint_secrets(request)
+    client = get_v3io_client(
+        endpoint=secrets[V3IO_API], access_key=secrets[V3IO_ACCESS_KEY]
+    )
     cursor = client.kv.new_cursor(
```
Comment (yaronha) [inline mlrun/api/api/endpoints/model_endpoints.py:157]: @Michaelliv would be better to init v3io client once, i believe there is a way to pass the user/access per request as well

Comment (Michaelliv) [reply mlrun/api/api/endpoints/model_endpoints.py:157]: Its kind of a singleton, so it inits once for the same set of params

File: mlrun/api/api/endpoints/model_endpoints.py (hunk 1 above)
Comment (Michaelliv) [reply mlrun/api/api/endpoints/model_endpoints.py:417]: @yaronha any thoughts about this one? I'm changing it as per @Hedingber's request, if you have any objections, please let me know.

**Review by Hedingber — APPROVED** [review]

**Pull request merged**